docker-java-transport-httpclient5/src/main/java/com/github/dockerjava/httpclient5/ApacheDockerHttpClientImpl.java

```diff
@@ -128,53 +138,27 @@ protected ApacheDockerHttpClientImpl(
             .build();
     }
 
-    private Registry<ConnectionSocketFactory> createConnectionSocketFactoryRegistry(
-        SSLConfig sslConfig,
-        URI dockerHost
+    private HttpClientConnectionOperator createConnectionOperator(
+        URI dockerHost,
+        SSLContext sslContext
     ) {
-        RegistryBuilder<ConnectionSocketFactory> socketFactoryRegistryBuilder = RegistryBuilder.create();
-
-        if (sslConfig != null) {
-            try {
-                SSLContext sslContext = sslConfig.getSSLContext();
-                if (sslContext != null) {
-                    socketFactoryRegistryBuilder.register("https", new SSLConnectionSocketFactory(sslContext));
-                }
-            } catch (Exception e) {
-                throw new RuntimeException(e);
-            }
-        }
-
-        return socketFactoryRegistryBuilder
-            .register("tcp", PlainConnectionSocketFactory.INSTANCE)
-            .register("http", PlainConnectionSocketFactory.INSTANCE)
-            .register("unix", new ConnectionSocketFactory() {
-                @Override
-                public Socket createSocket(HttpContext context) throws IOException {
-                    return UnixSocket.get(dockerHost.getPath());
-                }
-
-                @Override
-                public Socket connectSocket(TimeValue timeValue, Socket socket, HttpHost httpHost, InetSocketAddress inetSocketAddress,
-                                            InetSocketAddress inetSocketAddress1, HttpContext httpContext) throws IOException {
-                    return PlainConnectionSocketFactory.INSTANCE.connectSocket(timeValue, socket, httpHost, inetSocketAddress,
-                        inetSocketAddress1, httpContext);
-                }
-            })
-            .register("npipe", new ConnectionSocketFactory() {
-                @Override
-                public Socket createSocket(HttpContext context) {
+        String dockerHostScheme = dockerHost.getScheme();
+        String dockerHostPath = dockerHost.getPath();
+        TlsSocketStrategy tlsSocketStrategy = sslContext != null ?
+            new DefaultClientTlsStrategy(sslContext) : DefaultClientTlsStrategy.createSystemDefault();
+        return new DefaultHttpClientConnectionOperator(
+            socksProxy -> {
+                if ("unix".equalsIgnoreCase(dockerHostScheme)) {
+                    return UnixSocket.get(dockerHostPath);
+                } else if ("npipe".equalsIgnoreCase(dockerHostScheme)) {
                     return new NamedPipeSocket(dockerHost.getPath());
```

Comment:
[nit] Could use `dockerHostPath` here for consistency with the code two lines above:

```suggestion
                    return new NamedPipeSocket(dockerHostPath);
```

I don't feel strongly about this, so feel free to disregard this comment if you wish.